core/src/main/java/org/jruby/internal/runtime/NativeThread.java

```diff
@@ -109,4 +113,53 @@ public String toString() {
     public Thread nativeThread() {
         return nativeThread.get();
     }
+
+    @Override
+    public void setNativeName(String name) {
+        assert name != null;
+
+        Thread nativeThread = getThread();
+        if (nativeThread != null) {
+            try {
+                nativeThread.setName(name);
+            } catch (SecurityException se) {
+                // current thread cannot modify...ignore
+            }
+        }
+
+        this.nativeName = name;
+    }
+
+    @Override
+    public String getNativeName() {
+        return nativeName;
+    }
+
+    @Override
+    public void setRubyName(CharSequence name) {
+        this.rubyName = name;
+    }
+
+    @Override
+    public CharSequence getRubyName() {
+        return rubyName;
+    }
+
+    @Override
+    public String getReportName() {
+        String nativeName = getNativeName();
+
+        if (rubyName == null || rubyName.length() == 0) {
+            if (nativeName.equals("")) {
+                return "(unnamed)";
+            } else {
+                return nativeName;
+            }
+        } else {
+            if (nativeName.equals("")) {
+                return rubyName.toString();
```

Comment:
would add `'@' + rubyName` as well as bellow (so that we're consistent to have @ prefix for Ruby thread-names)